File: Tutorials/CS50_2025/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

float coleman_liau_index(string text);

int main(void)
{
    // Prompt the user for two words
    string text = get_string("Text :");

    // Apply Coleman-Liau index
    int grade = round(coleman_liau_index(text));

    // Print the output based on the grade
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", grade);
    }
}

float coleman_liau_index(string text)
{
    int letters = 0;
    int spaces = 0;
    int punctuations = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (isspace(text[i]))
        {
            spaces++;
        }
        else if (text[i] == '!' || text[i] == '.' || text[i] == '?')
        {
            punctuations++;
        }
    }
    int words = spaces + 1;
    int sentences = punctuations;

    float L = 100.0 * (float) letters / words;
    float S = 100.0 * (float) sentences / words;

    return 0.0588 * L - 0.296 * S - 15.8;
}

File: Tutorials/CS50_2025/sentimental-readability/readability.py
from cs50 import get_string


def coleman_liau_index(text):
    letters = 0
    punctuations = 0

    words = text.split(' ')
    for word in words:
        for char in word:
            if char.isalpha():
                letters += 1
            elif char in ['!', '?', '.']:
                punctuations += 1

    L = letters / len(words)
    S = punctuations / len(words)
    return 5.88 * L - 29.6 * S - 15.8


text = get_string("Text : ")

grade = coleman_liau_index(text)


if grade < 1:
    print("Before Grade 1")
elif (grade >= 16):
    print("Grade 16+")
else:
    print(f"Grade {round(grade)}")
